reference-controllers/rabbitmq-controller/src/main.rs

```diff
@@ -107,10 +142,171 @@ async fn reconcile_stateful_set(rabbitmq: &RabbitmqCluster, client: Client) -> R
             .create(&PostParams::default(), &sts)
             .await
             .map_err(Error::ReconcileStatefulSetFailed)?;
+        let gvk = GroupVersionKind::gvk("apps", "v1", "StatefulSet");
+        let api_resource = ApiResource::from_gvk(&gvk);
+        update_annotation(
+            api_resource,
+            rabbitmq.metadata.namespace.as_ref().unwrap(),
+            sts_name,
+            "createdAt",
+            &Utc::now().to_rfc3339(),
+            &client,
+        )
+        .await?;
+        Ok(())
+    }
+}
+
+async fn reconcile_server_config_map(
+    rabbitmq: &RabbitmqCluster,
+    client: &Client,
+) -> Result<(), Error> {
+    // Reconcile the RabbitmqCluster server configmap.
+    let cm_api = Api::<corev1::ConfigMap>::namespaced(
+        client.clone(),
+        rabbitmq.metadata.namespace.as_ref().unwrap(),
+    );
+    let server_config = server_configmap::server_configmap_build(&rabbitmq);
+    let server_config_name = server_config.metadata.name.as_ref().unwrap();
+    let server_config_o = cm_api
+        .get_opt(server_config_name)
+        .await
+        .map_err(Error::ReconcileConfigMapFailed)?;
+
+    if server_config_o.is_some() {
+        info!(
+            "Current rv of server configmap: {}",
+            server_config_o
+                .as_ref()
+                .unwrap()
+                .metadata
+                .resource_version
+                .as_ref()
+                .unwrap()
+        );
+        if server_config.data != server_config_o.as_ref().unwrap().data {
+            // If the data is different, update the server config
+            info!("Update configmap: {}", server_config_name);
+            info!(
+                "The new user config is: {}",
+                server_config.data.clone().unwrap()["userDefinedConfiguration.conf"]
+            );
+            let updated_server_config = ConfigMap {
+                data: server_config.data,
+                binary_data: server_config.binary_data,
+                immutable: server_config.immutable,
+                ..server_config_o.unwrap()
+            };
+            cm_api
+                .replace(
+                    server_config_name,
+                    &PostParams::default(),
+                    &updated_server_config,
+                )
+                .await
+                .map_err(Error::ReconcileConfigMapFailed)?;
+
+            let gvk = GroupVersionKind::gvk("", "v1", "ConfigMap");
+            let api_resource = ApiResource::from_gvk(&gvk);
+            update_annotation(
```

Comment:
Why do we need the annotation for configmap?

Reply:
It is used to store when did the configmap newly updated. We can delete it when we use resourceVerison(maybe next PR)